Extend bone override capabilities (#12388)
minetest/minetest: doc/lua_api.md

	* Shorthand for `get_bone_override(bone).position.vec, get_bone_override(bone).rotation.vec:apply(math.deg)`.
	* **Note:** Returned rotation is in degrees, not radians.
	* **Deprecated:** Use `get_bone_override` instead.
* `set_bone_override(bone, override)`
    * `bone`: string
    * `override`: `{ position = property, rotation = property, scale = property }` or `nil`
        * `property`: `{ vec = vector, interpolation = 0, absolute = false}` or `nil`;
            * `vec` is in the same coordinate system as the model, and in degrees for rotation
        * `property = nil` is equivalent to no override on that property
        * `absolute`: If set to `false`, the override will be relative to the animated property:
            * Transposition in the case of `position`;
            * Composition in the case of `rotation`;
            * Multiplication in the case of `scale`
        * `interpolation`: Old and new values are interpolated over this timeframe (in seconds)
    * `override = nil` (including omission) is shorthand for `override = {}` which clears the override
    * **Note:** Unlike `set_bone_position`, the rotation is in radians, not degrees.

>     * Compatibility note: Clients prior to 5.8.0 only support absolute position and rotation.

Reviewer: ```suggestion
    * Compatibility note: Clients prior to 5.9.0 only support absolute position and rotation.
```

    object.
* `set_detach()`: Detaches object. No-op if object was not attached.
* `set_bone_position([bone, position, rotation])`
	* Shorthand for `set_bone_override(bone, {position = position, rotation = rotation:apply(math.rad)})` using absolute values.

> 	* Supports older servers & clients (5.8.0 and earlier).

Reviewer 2: nitpick: this note seems to encourage using this function, however it is deprecated
that it supports old clients can also be deduced from below information

Author: Removed

	* **Note:** Rotation is in degrees, not radians.
	* **Deprecated:** Use `set_bone_override` instead.
* `get_bone_position(bone)`: returns the previously set position and rotation of the bone
	* Shorthand for `get_bone_override(bone).position.vec, get_bone_override(bone).rotation.vec:apply(math.deg)`.
	* **Note:** Returned rotation is in degrees, not radians.
	* **Deprecated:** Use `get_bone_override` instead.
* `set_bone_override(bone, override)`
    * `bone`: string
    * `override`: `{ position = property, rotation = property, scale = property }` or `nil`
        * `property`: `{ vec = vector, interpolation = 0, absolute = false}` or `nil`;
            * `vec` is in the same coordinate system as the model, and in degrees for rotation
        * `property = nil` is equivalent to no override on that property
        * `absolute`: If set to `false`, the override will be relative to the animated property:
            * Transposition in the case of `position`;
            * Composition in the case of `rotation`;
            * Multiplication in the case of `scale`

>         * `interpolation`: Time it takes (on the client, in seconds) to interpolate between the old & new value

Reviewer 2: dunno
```suggestion
        * `interpolation`: Old and new values are interpolated over this timeframe (in seconds)
```
